test/Test/Util/QC.hs

```diff
@@ -15,3 +18,16 @@ testClassLaws typename Laws {lawsTypeclass, lawsProperties} =
     | (name, prop) <- lawsProperties ]
 
 
+-- | A 'Choice' of a uniform random number in a range where shrinking picks smaller numbers.
+newtype Choice = Choice Word64
+  deriving stock (Show, Eq)
+  deriving newtype (Arbitrary)
+
+getChoice :: (Integral a) => Choice -> (a, a) -> a
+getChoice (Choice n) (l, u) = fromIntegral (((ni * (ui - li)) `div` mi) + li)
```

Comment:
Can we expect for all `Integral a` that the `fromIntegral` conversion preserves uniformity? We should maybe add a comment about it

Reply:
Right, it's uniform as long as the range isn't larger than the largest `Word64`.